docs/fsharp-cheatsheet.md

```diff
@@ -169,197 +205,281 @@ The `rec` keyword is used together with the `let` keyword to define a recursive
+    // Indexed access
+    let first = array1[0]  // "a"
 
-    // Sequences can use yield and contain subsequences
-    let seq1 =
-        seq {
-            // "yield" adds one element
-            yield 1
-            yield 2
+    // Slicing is inclusive; [| "a"; "b" |]
+    let firstTwo = array1[0..1]
 
-            // "yield!" adds a whole subsequence
-            yield! [5..10]
-        }
+    // Assignment using `<-`
+    array1[1] <- "d"  // [| "a"; "d"; "c" |]
 
-### Higher-order functions on collections
-The same list `[ 1; 3; 5; 7; 9 ]` or array `[| 1; 3; 5; 7; 9 |]` can be generated in various ways.
+    // Pattern matching
+    match myArray with
+    | [| |] -> ...       // match an empty array
+    | [| 3 |] -> ...     // match array with single 3 item
+    | [| _; 4 |] -> ...  // match array with 2 items, second item = 4
 
- - Using range operator `..`
+See the [Array Module](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-collections-arraymodule.html) for built-in functions.
 
-        let xs = [ 1..2..9 ]
+<div id="collections-sequences"></div>
 
- - Using list or array comprehensions
+### Sequences
 
-        let ys = [| for i in 0..4 -> 2 * i + 1 |]
+A *sequence* is a logical series of elements of the same type. Individual sequence elements are computed only as required,
+so a sequence can provide better performance than a list in situations in which not all the elements are used.
 
- - Using `init` function
+    // Create
+    let seq1 = { 1; 2 }
+    let seq2 =
+        { 1
+          2 }
+    let seq3 = seq { 1..2..9 }  // start..skip..last; 1,3,5,7,9
 
-        let zs = List.init 5 (fun i -> 2 * i + 1)
+See the [Seq Module](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-collections-seqmodule.html) for built-in functions.
 
-Lists and arrays have comprehensive sets of higher-order functions for manipulation.
+### Higher-order functions on collections
 
-  - `fold` starts from the left of the list (or array) and `foldBack` goes in the opposite direction
+- Computed expressions with `->`. Results in _1, 3, 5, 7, 9_
 
-        let xs' =
-            Array.fold (fun str n ->
-                sprintf "%s,%i" str n) "" [| 0..9 |]
+       let listComp  = [ for i in 0..4 -> 2 * i + 1 ]
+       let arrayComp = [| for i in 0..4 -> 2 * i + 1 |]
+       let seqComp   = seq { for i in 0..4 -> 2 * i + 1 }
 
-  - `reduce` doesn't require an initial accumulator
+- Using computed expressions with `yield` and `yield!`:
 
-        let last xs = List.reduce (fun acc x -> x) xs
+        let listWithYield = [  // [ 1;3;5;7;9 ]
+            for i in 0..4 do
+                yield 2 * i + 1
+            ]
+        let arrayWithYield = [|  // [| 1;3;5;7;9;1;3;5;7;9 |]
+            for i in 0..4 do
+                yield 2 * i + 1
```

Comment:
mention or illustrate that `yield` is optional in a `do`

Reply:
maybe do that first and remove the `withYield` from the names
also this is not really higher order fuctions on collections - its comprehensions